third_party/closure-responding-channel/compile.sh

```diff
@@ -2,44 +2,30 @@
 
 # install google closure library:  `npm install google-closure-library` at amphtml
 # folder
-CLOSURE_LIB=../../../../node_modules/google-closure-library
+CLOSURE_LIB=../../node_modules/google-closure-library
 
-# TODO: ensure the new up-to-date compiler could generate the same output as the Google's internal one
-# after the PR to update Closure compiler got in(https://github.com/ampproject/amphtml/pull/33153). Currently
-# the output of this script is not used.
 npx google-closure-compiler \
-  --jscomp_warning=nonStandardJsDocs \
-  --jscomp_warning=checkDebuggerStatement \
-  --jscomp_warning=externsValidation \
-  --jscomp_warning=missingOverride \
-  --jscomp_error=checkRegExp \
-  --jscomp_error=checkTypes \
-  --jscomp_error=const \
-  --jscomp_error=es5Strict \
-  --jscomp_error=globalThis \
-  --jscomp_error=missingProvide \
-  --jscomp_error=missingProperties \
-  --jscomp_error=missingReturn \
-  --jscomp_error=nonStandardJsDocs \
-  --jscomp_error=suspiciousCode \
-  --jscomp_error=tooManyTypeParams \
-  --jscomp_error=unknownDefines \
-  --jscomp_error=uselessCode \
-  --jscomp_error=visibility \
-  --language_out=ECMASCRIPT5_STRICT \
   --define=goog.DEBUG=false \
-  --generate_exports \
-  --use_types_for_optimization \
   --compilation_level ADVANCED_OPTIMIZATIONS \
-  --define=goog.TRANSPILE=never \
-  --define=goog.LOAD_MODULE_USING_EVAL=false \
-  --js_output_file "closure-bundle.js" \
+  --formatting=PRETTY_PRINT \
   --output_wrapper "
```

Comment:
We still need to wrap the output in a closure, so that it's not referencing the real `window`:

```
// Create a sub-object that inherits from window, so that `window.foo = 1` mutates our local object,
// and `window.bar` can still receive all globals.
const local = Object.create(window);
(function(window) {
  %output%;
}).call(local, local
export function createPortChannel(frameWindow, origin) {
  return local.__AMP_createPortChannel(frameWindow, origin);
}
// ...
```

Reply:
Done. One tweak: call(local, local) has to be chaned to call(window, local), otherwise some native function calls that need to be happend within the context of real window.

Reply:
But that's the thing we're trying to avoid. Passing in `window` as the first parameter leads to `exportSymbol` adding a property to it. Passing in `localWindow` is meant to give a mutable reference object, though which things like `local.Array` or `local.Uint8Array` will still work (because of the prototype setup done by `Object.create`).

Reply:
Hmm I thought the second parameter is the mutable reference object, no?

When using in call(local, local), I am having the "Uncaught TypeError: Illegal invocation"(similar to [this issue](https://stackoverflow.com/questions/9677985/uncaught-typeerror-illegal-invocation-in-chrome)). It does seem like it needs a window object to function. Let me try to see why Object.create is not working properly to set up the right properties of a window object.